Service/ExpressPay/QuoteConverter.php

```diff
@@ -230,16 +238,22 @@ public function convertQuoteItems(Quote $quote): array
         }
 
         $currencyCode = $quote->getCurrency() !== null ? $quote->getCurrency()->getQuoteCurrencyCode() : '';
+        $websiteId = (int)$quote->getStore()->getWebsiteId();
+        $taxIncluded = $this->config->isTaxIncludedInPrices($websiteId);
         $convertedQuote = [
             'order_data' => [
                 'items' => array_map(
-                    static function (CartItemInterface $cartItem) use ($currencyCode): array {
+                    static function (CartItemInterface $cartItem) use ($taxIncluded, $currencyCode): array {
+                        $itemData = $cartItem->getData();
+
+                        $price = $taxIncluded ? $itemData['base_price_incl_tax'] - $itemData['tax_amount'] : $cartItem->getPrice();
```

Comment:
Is `$cartItem->getPrice()` guaranteed to be a number here? Previously it looks like we were casting this to a float, not sure if we want/need to continue to do that.

Reply:
Nice catch! It should always be a number, but I carried over the casting to keep the previous functionality as close as possible